### PR TITLE
Addition of integration test

### PR DESCRIPTION
**Pull request by aalvarado1**
This also includes some updates to docker file with regards to x509 issue in dev kube cluster. Updates to include viper for test configurations and types for kube client.

### CODE REVIEW

File: main_test.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+func TestUnitExample(t *testing.T) {
+	//Test example
+	fmt.Println("Unit test 1")
+	actual := strings.ToUpper("hello")
+	expected := "HELLO"
+
+	assert.Equal(t, expected, actual)
+}
+
```
Comment (ctriv) [inline main_test.go:25]: Probably should remove this test 😉

File: main_test.go [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+	err := client.c.CoreV1().Secrets("default").Delete(testCert.Spec.SecretName, nil)
+
+	if err != nil {
+		log.Printf("Error while deleting secret: %v.", err)
+	}
+}
+
+func getRandomNumber() int {
+	rand.Seed(time.Now().UnixNano())
```
Comment (ctriv) [inline main_test.go]: Probably don't want to reseed the RNG each time...  maybe an Init() function?

**Review by ctriv — APPROVED** [review]

File: main_test.go (hunk 1 above)
Comment (aalvarado1) [reply main_test.go:25]: 👍

File: main_test.go (hunk 2 above)
Comment (aalvarado1) [reply main_test.go]: Sounds good

**Pull request merged**